pkg/flow/internal/controller/node_declare.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/grafana/river/ast"
+	"github.com/grafana/river/vm"
+)
+
+type DeclareNode struct {
+	label         string
+	nodeID        string
```

Comment:
Isn't nodeID redundant? it seems to be always `label + componentName`.

Reply:
it's `componentName+"."+label`. This is common to every node and it is used for optimization (from builtin component node: "nodeID string // Cached from id.String() to avoid allocating new strings every time NodeID is called.")